### ISSUE TITLE
[quarkus-main] MyBatis integration tests fail with error creating synthetic bean

### ISSUE THREAD

**Issue by jamesnetherton** [apache/camel-quarkus#5656]
### Bug description

FYI @zhfeng 

```
2024-01-19T03:43:32.7899140Z [INFO] Running org.apache.camel.quarkus.component.mybatis.it.MyBatisConsumerTest
2024-01-19T03:43:38.3812775Z [INFO] H2 database started in TCP server mode; server status: TCP server running at tcp://10.1.0.131:9092 (only local connections)
2024-01-19T03:43:39.1832471Z 2024-01-19 03:43:39,165 INFO  [org.apa.cam.uti.con.ThreadType] (main) The type of thread detected is PLATFORM
2024-01-19T03:43:39.4839991Z 2024-01-19 03:43:39,479 WARN  [io.qua.myb.run.MyBatisRecorder] (main) Error executing SQL Script insert.sql: jakarta.enterprise.inject.CreationException: Error creating synthetic bean [svmyPJfgLUOBzT7zpcDjIrSgwlc]: jakarta.enterprise.inject.CreationException: Synthetic bean instance for io.quarkus.datasource.runtime.DataSourceSupport not initialized yet: io_quarkus_datasource_runtime_DataSourceSupport_f2cd6de2f5ebdf7a97f46dda37f1938b9d43bde4
2024-01-19T03:43:39.4844601Z 	- a synthetic bean initialized during RUNTIME_INIT must not be accessed during STATIC_INIT
2024-01-19T03:43:39.4846719Z 	- RUNTIME_INIT build steps that require access to synthetic beans initialized during RUNTIME_INIT should consume the SyntheticBeansRuntimeInitBuildItem
2024-01-19T03:43:39.4849211Z 	at io.quarkus.datasource.runtime.DataSourceSupport_svmyPJfgLUOBzT7zpcDjIrSgwlc_Synthetic_Bean.doCreate(Unknown Source)
2024-01-19T03:43:39.4851188Z 	at io.quarkus.datasource.runtime.DataSourceSupport_svmyPJfgLUOBzT7zpcDjIrSgwlc_Synthetic_Bean.create(Unknown Source)
2024-01-19T03:43:39.4853128Z 	at io.quarkus.datasource.runtime.DataSourceSupport_svmyPJfgLUOBzT7zpcDjIrSgwlc_Synthetic_Bean.create(Unknown Source)
2024-01-19T03:43:39.4854961Z 	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
2024-01-19T03:43:39.4856533Z 	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
2024-01-19T03:43:39.4857924Z 	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
2024-01-19T03:43:39.4859092Z 	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:32)
2024-01-19T03:43:39.4860255Z 	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
2024-01-19T03:43:39.4862068Z 	at io.quarkus.arc.impl.ComputingCacheContextInstances.computeIfAbsent(ComputingCacheContextInstances.java:19)
2024-01-19T03:43:39.4863815Z 	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
2024-01-19T03:43:39.4865498Z 	at io.quarkus.datasource.runtime.DataSourceSupport_svmyPJfgLUOBzT7zpcDjIrSgwlc_Synthetic_Bean.get(Unknown Source)
2024-01-19T03:43:39.4867474Z 	at io.quarkus.datasource.runtime.DataSourceSupport_svmyPJfgLUOBzT7zpcDjIrSgwlc_Synthetic_Bean.get(Unknown Source)
2024-01-19T03:43:39.4869049Z 	at io.quarkus.agroal.runtime.DataSources_Bean.doCreate(Unknown Source)
2024-01-19T03:43:39.4870278Z 	at io.quarkus.agroal.runtime.DataSources_Bean.create(Unknown Source)
2024-01-19T03:43:39.4871448Z 	at io.quarkus.agroal.runtime.DataSources_Bean.create(Unknown Source)
2024-01-19T03:43:39.4872911Z 	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
2024-01-19T03:43:39.4874501Z 	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
2024-01-19T03:43:39.4875905Z 	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
2024-01-19T03:43:39.4877069Z 	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:32)
2024-01-19T03:43:39.4878240Z 	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
2024-01-19T03:43:39.4879956Z 	at io.quarkus.arc.impl.ComputingCacheContextInstances.computeIfAbsent(ComputingCacheContextInstances.java:19)
2024-01-19T03:43:39.4881889Z 	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
2024-01-19T03:43:39.4883180Z 	at io.quarkus.agroal.runtime.DataSources_Bean.get(Unknown Source)
2024-01-19T03:43:39.4884461Z 	at io.quarkus.agroal.runtime.DataSources_Bean.get(Unknown Source)
2024-01-19T03:43:39.4885787Z 	at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:554)
2024-01-19T03:43:39.4887330Z 	at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:534)
2024-01-19T03:43:39.4889137Z 	at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:567)
2024-01-19T03:43:39.4890843Z 	at io.quarkus.arc.impl.ArcContainerImpl.instanceHandle(ArcContainerImpl.java:529)
2024-01-19T03:43:39.4892259Z 	at io.quarkus.arc.impl.ArcContainerImpl.instance(ArcContainerImpl.java:294)
2024-01-19T03:43:39.4893707Z 	at io.quarkus.agroal.runtime.DataSources.fromName(DataSources.java:127)
2024-01-19T03:43:39.4895218Z 	at io.quarkiverse.mybatis.runtime.QuarkusDataSource.getDataSource(MyBatisRecorder.java:453)
2024-01-19T03:43:39.4896882Z 	at io.quarkiverse.mybatis.runtime.QuarkusDataSource.getConnection(MyBatisRecorder.java:460)
2024-01-19T03:43:39.4898607Z 	at org.apache.ibatis.transaction.managed.ManagedTransaction.openConnection(ManagedTransaction.java:89)
2024-01-19T03:43:39.4900504Z 	at org.apache.ibatis.transaction.managed.ManagedTransaction.getConnection(ManagedTransaction.java:60)
2024-01-19T03:43:39.4902350Z 	at org.apache.ibatis.session.defaults.DefaultSqlSession.getConnection(DefaultSqlSession.java:297)
2024-01-19T03:43:39.4904038Z 	at io.quarkiverse.mybatis.runtime.MyBatisRecorder.runInitialSql(MyBatisRecorder.java:430)
20
```

**Comment by zhfeng** [comment]
@jamesnetherton It should be fixed in quarkus-mybatis `2.2.1`

- https://github.com/apache/camel-quarkus/pull/5664

**Comment by jamesnetherton** [comment]
Seems to be fixed on the latest quarkus-main nightly build. Thanks @zhfeng.
